### ISSUE TITLE
Issue with loading workers

### ISSUE THREAD

**Issue by rajialtooro** [suren-atoyan/monaco-react#541]
**Describe the bug**
I am facing an issue with loading the editor loaders (css, js, ts , html...)
I can see that the worker has been installed in the browser, but there is no auto-complete or text highlights

**To Reproduce**
I made a fresh Nextjs App
creates a file called `editor.tsx`
```ts
'use client'
import Editor, {loader} from '@monaco-editor/react';
import * as monaco from 'monaco-editor'
loader.config({monaco})

window.MonacoEnvironment = {
    getWorker: async function (moduleId, label) {
      console.log(moduleId, label)
      switch (label) {
        case 'editorWorkerService':
          return new Worker(
            new URL('monaco-editor/esm/vs/editor/editor.worker', import.meta.url)
          )
        case 'css':
        case 'less':
        case 'scss':
          return new Worker(
            new URL(
              'monaco-editor/esm/vs/language/css/css.worker',
              import.meta.url
            )
          )
        case 'handlebars':
        case 'html':
        case 'razor':
          return new Worker(
            new URL(
              'monaco-editor/esm/vs/language/html/html.worker',
              import.meta.url
            )
          )
        case 'json':
          return new Worker(
            new URL(
              'monaco-editor/esm/vs/language/json/json.worker',
              import.meta.url
            )
          )
        case 'javascript':
        case 'typescript':
          return new Worker(
            new URL(
              'monaco-editor/esm/vs/language/typescript/ts.worker',
              import.meta.url
            )
          )
        default:
          throw new Error(`Unimplemented worker ${label} (${moduleId})`)
      }
    },
  }
  

const TestEditor =() => {
  return <Editor height="90vh" defaultLanguage="css" defaultValue="// some comment" />;
}

export default TestEditor;
```
**Screenshots**
![image](https://github.com/suren-atoyan/monaco-react/assets/88225432/40687bb0-5da8-4a1c-b1ad-7407b9b04ce1)

![image](https://github.com/suren-atoyan/monaco-react/assets/88225432/5acb033a-faab-4c0d-ba19-dd8c73fe0f3b)
as you can see, there is no auto-complete or highlights

any ideas how I can load the workers from the cdn that used to work perfectly instead of getting the local ones ?

**Comment by AlpayC** [comment]
> **Describe the bug** I am facing an issue with loading the editor loaders (css, js, ts , html...) I can see that the worker has been installed in the browser, but there is no auto-complete or text highlights
> 
> **To Reproduce** I made a fresh Nextjs App creates a file called `editor.tsx`
> 
> ```ts
> 'use client'
> import Editor, {loader} from '@monaco-editor/react';
> import * as monaco from 'monaco-editor'
> loader.config({monaco})
> 
> window.MonacoEnvironment = {
>     getWorker: async function (moduleId, label) {
>       console.log(moduleId, label)
>       switch (label) {
>         case 'editorWorkerService':
>           return new Worker(
>             new URL('monaco-editor/esm/vs/editor/editor.worker', import.meta.url)
>           )
>         case 'css':
>         case 'less':
>         case 'scss':
>           return new Worker(
>             new URL(
>               'monaco-editor/esm/vs/language/css/css.worker',
>               import.meta.url
>             )
>           )
>         case 'handlebars':
>         case 'html':
>         case 'razor':
>           return new Worker(
>             new URL(
>               'monaco-editor/esm/vs/language/html/html.worker',
>               import.meta.url
>             )
>           )
>         case 'json':
>           return new Worker(
>             new URL(
>               'monaco-editor/esm/vs/language/json/json.worker',
>               import.meta.url
>             )
>           )
>         case 'javascript':
>         case 'typescript':
>           return new Worker(
>             new URL(
>               'monaco-editor/esm/vs/language/typescript/ts.worker',
>               import.meta.url
>             )
>           )
>         default:
>           throw new Error(`Unimplemented worker ${label} (${moduleId})`)
>       }
>     },
>   }
>   
> 
> const TestEditor =() => {
>   return <Editor height="90vh" defaultLanguage="css" defaultValue="// some comment" />;
> }
> 
> export default TestEditor;
> ```
> 
> **Screenshots** ![image](https://user-images.githubusercontent.com/88225432/271329759-40687bb0-5da8-4a1c-b1ad-7407b9b04ce1.png)
> 
> ![image](https://user-images.githubusercontent.com/88225432/271330031-5acb033a-faab-4c0d-ba19-dd8c73fe0f3b.png) as you can see, there is no auto-complete or highlights
> 
> any ideas how I can load the workers from the cdn that used to work perfectly instead of getting the local ones ?

Yeah . This happens with the prop defaultLanguage, too. I use onChange Events to change the defaultLanguage. Even with a useEffect it highlight the syntax not correctly. Here is the workaround. Rerender the Editor Component like this, and it will load the autocompletion and the highlights:


`  const [outputEditorKey, setOutputEditorKey] = useState(0);
`

`useEffect(() => {
    setOutputEditorKey((prevKey) => prevKey + 1);
  }, [outputTech]);
`

`const TestEditor =() => {
  return <Editor  key={`outputEditor-${outputEditorKey}`} height="90vh" defaultLanguage="css" defaultValue="// some comment" />;
}`
